sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -68,6 +68,7 @@ abstract class Optimizer extends RuleExecutor[LogicalPlan] {
       PushPredicateThroughAggregate,
       ColumnPruning,
       // Operator combine
+      RepartitionCollapsing,
```

Comment:
this was in the existing code, but it seems like we should call this CollapseRepartition, and the next one CollapseProject